src/com/lftechnology/java/training/krishna/io/countcharacter/CountCharacter.java

```diff
@@ -0,0 +1,96 @@
+
+package com.lftechnology.java.training.krishna.io.countcharacter;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This program count the no. of character occurrence on given file
+ * 
+ * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+ * @version 1.0
+ */
+public class CountCharacter {
+
+	private static final Logger LOGGER =
+		Logger.getLogger(CountCharacter.class.getName());
+
+	private String filePath;
+	private char inputCharacter;
+
+	public CountCharacter(String filePath, char inputCharacter) {
+
+		this.filePath = filePath;
+		this.inputCharacter = inputCharacter;
+	}
+
+	public String getFilePath() {
+
+		return filePath;
+	}
+
+	public void setFilePath(String filePath) {
+
+		this.filePath = filePath;
+	}
+
+	public char getInputCharacter() {
+
+		return inputCharacter;
+	}
+
+	public void setInputCharacter(char inputCharacter) {
+
+		this.inputCharacter = inputCharacter;
+	}
+
+	@Override
+	public String toString() {
+
+		return "CountCharacter [filePath=" + filePath + ", inputCharacter=" +
+			inputCharacter + "]";
+	}
+
+	/**
+	 * This method computes the no. of character occurrence.
+	 * 
+	 * @return count {@link int} no. of character occurrence
+	 * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+	 */
+	public int countCharacterOccurance() {
+
+		BufferedReader reader = null;
+		int count = 0;
+		try {
+			Reader file = new FileReader(getFilePath());
+			reader = new BufferedReader(file);
+			String line = null;
+			while ((line = reader.readLine()) != null) {
+				for (int i = 0; i < line.length(); i++) {
+					if (inputCharacter == line.charAt(i)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS134)